modules/haloRtdProvider.js

```diff
@@ -1,109 +1,102 @@
   }
+  if (!isPlainObject(source)) {
+    source = {};
+  }
+  return mergeDeep(target, source);
 }
 
 /**
- * decorate adUnits with segment data
- * @param {adUnit[]} adUnits
- * @param {Object} data
+ * Param or default.
+ * @param {String} param
+ * @param {String} defaultVal
  */
-export function addSegmentData(adUnits, segmentData, config) {
-  adUnits.forEach(adUnit => {
-    if (adUnit.hasOwnProperty('bids')) {
-      adUnit.bids.forEach(bid => {
-        try {
-          set(bid, 'fpd.user.data', []);
-          if (Array.isArray(bid.fpd.user.data)) {
-            bid.fpd.user.data.forEach(fpdData => {
-              let segments = segmentData[fpdData.id] || segmentData[fpdData.name] || [];
-              fpdData.segment = (fpdData.segment || []).concat(segments);
-            });
-          }
-        } catch (err) {
-          utils.logError(err.message);
-        }
+function paramOrDefault(param, defaultVal) {
+  if (isFn(param)) {
+    return param();
+  } else if (isStr(param)) {
+    return param;
+  }
+  return defaultVal;
+}
 
-        try {
-          if (config.params.mapSegments && config.params.mapSegments[bid.bidder] && segmentData[bid.bidder]) {
-            if (typeof config.params.mapSegments[bid.bidder] == 'function') {
-              config.params.mapSegments[bid.bidder](bid, segmentData[bid.bidder]);
-            } else if (segmentMappers[bid.bidder]) {
-              segmentMappers[bid.bidder](bid, segmentData[bid.bidder]);
-            }
-          }
-        } catch (err) {
-          utils.logError(err.message);
-        }
-      });
-    }
-  });
+/**
+ * Add real-time data & merge segments.
+ * @param {Object} bidConfig
+ * @param {Object} rtd
+ * @param {Object} rtdConfig
+ */
+export function addRealTimeData(bidConfig, rtd, rtdConfig) {
+  if (!isPlainObject(config.getConfig().ortb2)) {
+    config.setConfig({ortb2: {}});
+  }
 
-  return adUnits;
+  if (rtdConfig.params && rtdConfig.params.handleRtd) {
+    rtdConfig.params.handleRtd(bidConfig, rtd, rtdConfig, config);
+  } else if (rtd.ortb2) {
+    let mergedOrtb = mergeLazy(config.getConfig().ortb2, rtd.ortb2);
+    config.setConfig({ortb2: mergedOrtb});
+  }
 }
 
 /**
- * segment retrieval from audigent's backends
+ * Real-time data retrieval from Audigent
  * @param {Object} reqBidsConfigObj
  * @param {function} onDone
- * @param {Object} config
+ * @param {Object} rtdConfig
  * @param {Object} userConsent
  */
-export function getSegments(reqBidsConfigObj, onDone, config, userConsent) {
-  const adUnits = reqBidsConfigObj.adUnits || getGlobal().adUnits;
-
-  if (config.params.segmentCache) {
-    let jsonData = storage.getDataFromLocalStorage(SEG_LOCAL_NAME);
+export function getRealTimeData(bidConfig, onDone, rtdConfig, userConsent) {
+  if (isPlainObject(rtdConfig.params) && rtdConfig.params.segmentCache) {
+    let jsonData = storage.getDataFromLocalStorage(RTD_LOCAL_NAME);
 
     if (jsonData) {
       let data = JSON.parse(jsonData);
```

Comment:
Maybe surround in try catch just in case? Outside of this PR change so feel free to ignore